### PR TITLE
Introduce a new WebKit checker for a unchecked local variable

### PR DESCRIPTION
**Pull request by rniwa**
This PR introduces alpha.webkit.UncheckedLocalVarsChecker which detects a raw reference or a raw pointer local, static, or global variable to a CheckedPtr capable object without a guardian variable in an outer scope.

### CODE REVIEW

File: clang/test/Analysis/Checkers/WebKit/unchecked-local-vars.cpp [hunk 1]
```diff
@@ -0,0 +1,342 @@
...
+  }
+}
+
+void static_var() {
+  static RefCountableAndCheckable* a = nullptr;
+  // expected-warning@-1{{Static local variable 'a' is unchecked and unsafe [alpha.webkit.UncheckedLocalVarsChecker]}}
+  a = provide_obj();
+}
+
+} // namespace local_refcountable_checkable_object
```
Comment (t-rasmud) [inline clang/test/Analysis/Checkers/WebKit/unchecked-local-vars.cpp:342]: Do we expect a similar warning for assignments to struct members or class fields?

Comment (rniwa) [reply clang/test/Analysis/Checkers/WebKit/unchecked-local-vars.cpp:342]: No, struct members and class fields are checked by `NoUncountedMemberChecker` / `NoUncheckedPtrMemberChecker`.

**Review by t-rasmud — APPROVED** [review]
LGTM!

**Comment by rniwa** [comment]
> LGTM!

Thanks for the review!

**Pull request merged**